### PR TITLE
Fix timeout exceptions when using tcnative

### PR DESCRIPTION
**Pull request by xinlian12**
**Issue**: https://github.com/Azure/azure-sdk-for-java/issues/7829

**Root cause**: 
There are some implementation difference between OpenSslEngine and JDKSslEngine. When enabling tcnative-boringssl, what has been found is that in the inbound bytebuf, there could be several responses unwrapped. So this issue is caused by combination of  multipleResponseUnwrapped + discardReadBytes + using of slice of the bytebuf. 
1) multiple response is unwrapped in the same bytebuf (A, B, C)
2) RntbdResponse.decode-> slice of the bytebuf (which is a view of the subregion)
3) RntbdResponse.discardReadBytes -> reset the readerIndex/writerIndex of the bytebuf. This might involve extra memory copy around.
4) RntbdRequestManager.getTransportRequestId -> The transportRequestId returned is B's transportRequestId , complete B, A got requestTimeoutException.

**Possible Solutions**:
1) Remove discardReadBytes
2) Enable jdkCompatibilityMode -> this PR
3) Do not use slice of the bytebuf

**Test Results**:
image
image
image
image

### CODE REVIEW

**Review by moderakh — CHANGES_REQUESTED** [review]
great work Annie. 

This issue which was discovered on v2 has been bugging us for a very long time. Very happy to see this finally fixed. 

There is an exclusion in azure-cosmos pom file. please remove that too.

File: sdk/cosmos/azure-cosmos-benchmark/pom.xml [hunk 1]
```diff
@@ -88,14 +88,11 @@ Licensed under the MIT License.
       <version>1.5.4</version> <!-- {x-version-update;io.micrometer:micrometer-registry-graphite;external_dependency} -->
     </dependency>
 
-    <!-- Issue: We have a problem with Direct TCP with using netty-tcnative when OpenSSL 1.1 is installed
     <dependency>
       <groupId>io.netty</groupId>
-      <artifactId>netty-tcnative</artifactId>
-      <version>2.0.31.Final</version> {x-version-update;io.netty:netty-tcnative;external_dependency}
-      <classifier>linux-x86_64</classifier>
+      <artifactId>netty-tcnative-boringssl-static</artifactId>
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/pom.xml:93]: please update azure-cosmos and remove the explicit exclusion. This library is a dependency of azure-core we don't need to have explicit dependency on it.

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-benchmark/pom.xml]: Thanks Mo~  I removed the exclusions from azure-cosmos and azure-cosmos-encryption

**Review by moderakh — APPROVED** [review]
fantastic. Thank you Annie.

LGTM.

**Comment by xinlian12** [comment]
Tested the following test locally and passed:
queryMetricsForReadAllItemsOfLogicalPartition

**Review by David-Noble-at-work — APPROVED** [review]
LGTM with one question outstanding question. Approved assuming there's not work to do to ensure that the benchmark program runs with netty-tcnative, ideally netty-tcnative-boringssl-static.

File: sdk/cosmos/azure-cosmos-benchmark/pom.xml [hunk 2]
```diff
@@ -88,15 +88,6 @@ Licensed under the MIT License.
       <version>1.5.4</version> <!-- {x-version-update;io.micrometer:micrometer-registry-graphite;external_dependency} -->
     </dependency>
```
[the remark this answers is not in the document]
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-benchmark/pom.xml:90]: It should get brought in through azure-core.

@xinlian12  could you please run
```bash
mvn depedency:tree
```
to verify

[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-benchmark/pom.xml:90]: yea, boring ssl will be brought in by azure-core:
image

**Pull request merged**